src/main/java/com/emc/mongoose/storage/driver/pravega/PravegaStorageDriver.java

```diff
@@ -173,7 +176,14 @@ private void invokeFileNio(DataOperation<? extends DataItem> fileOperation) {
 					//TODO: EventStreamWriter<ByteBuffer>  ->  writeEvent
 					break;
 				case READ:
-					//TODO: EventStreamReader<ByteBuffer>  ->  readNextEvent
+					if ((null != scopeMap.get(scopeName) && null != scopeMap.get(scopeName).get(streamName))) {
+						if (invokeDataRead(fileOperation, scopeName, streamName)) {
+						}
+					} else {
+						Loggers.ERR.debug(
+								"Failed to read the stream {} in the scope {}", streamName, scopeName);
```

Comment:
as we discussed with Andrey, we don't need to check scope and stream before reading. We just need to correctly react if pravega falls due to absent scope or stream.